CMakeLists.txt

```diff
@@ -18,6 +18,15 @@ option(CCTAG_USE_POSITION_INDEPENDENT_CODE "Generate position independent code."
 option(CCTAG_ENABLE_SIMD_AVX2 "Enable AVX2 optimizations" OFF)
 option(BUILD_TESTS "Build the unity tests" ON)
 
+# Set a default build type if none was specified
```

Comment:
You can do it the same way that what is done in AliceVision:
https://github.com/alicevision/AliceVision/blob/develop/src/CMakeLists.txt#L78

Reply:
On Windows, as it generates the MSVC project which contains both release and debug, the CMAKE_BUILD_TYPE is not defined.

Reply:
I've done that and pushed it.

Reply:
What about something like that
'''
# Set a default build type if none was specified
set(default_build_type "Release")

if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
    message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
    set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE  STRING "Choose the type of build." FORCE)
    # Set the possible values of build type for cmake-gui
    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS  "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
endif()
'''

Plus the mvc stuff

Reply:
It's more code and more difficult to read, IMHO. Is there a difference in term of behavior?

I was suggesting to use the code from AV, as we are using it daily on win and linux without issue.
If we can do it better, we should update AV, CCTag, PopSift.

Reply:
I like to be able to switch between Release and Debug modes without removing the CMake cache, so for me, the 3 simple lines from AliceVision are the better option.